xbmc/music/MusicDatabase.cpp

```diff
@@ -502,7 +502,17 @@ bool CMusicDatabase::AddAlbum(CAlbum& album, int idSource)
       PrepareSQL("UPDATE album SET iDisctotal = (SELECT COUNT(DISTINCT iTrack >> 16) FROM song "
                  "WHERE song.idAlbum = album.idAlbum) WHERE idAlbum = %i",
                  album.idAlbum));
-
+  std::string strSQL;
+  strSQL = PrepareSQL(
+      "SELECT COUNT(DISTINCT strDiscSubtitle) FROM song WHERE song.idAlbum = %i", album.idAlbum);
+  int numTitles = strtol(GetSingleValue(strSQL).c_str(), nullptr, 10);
+  if (!album.bBoxedSet) // not already a boxset via tag
+    if (!album.bCompilation) // and not a compilation album
+      if (numTitles >= 3) // 3 distinct titles already so this must be a boxset
```

Comment:
Done like this needs only one if statement 
`If (!album.bBoxedSet && !album.bCompilation && numTitles >= 3)`
But you could move the `SELECT COUNT(DISTINCT strDiscSubtitle) ... ` query inside a check that `!album.bBoxedSet && !album.bCompilation` -  only do query when necessary